ui/src/app/shared/services/workflows-service.ts

```diff
@@ -241,22 +241,43 @@ export const WorkflowsService = {
                     }
                     throw new Error('no artifact logs are available');
                 }
-
                 return from(requests.get(this.getArtifactLogsPath(workflow, nodeId, container, archived)));
             }),
             mergeMap(r => r.text.split('\n')),
+            filter(x => !!x), // check for empty content
             map(content => ({content, podName: workflow.status.nodes[nodeId].displayName}) as LogEntry),
             filter(x => !!x.content.match(grep))
         );
     },
 
+    getContainerLogsFromArtifacts(workflow: Workflow, container: string, grep: string, archived: boolean): Observable<LogEntry> {
+        // Iterate over all nodes and check if node is a pod. Then sort nodes by startedAt.
+        const podNodeIds: string[] = Object.keys(workflow.status.nodes)
+            .filter(nodeId => {
+                return workflow.status.nodes[nodeId].type === 'Pod';
+            })
+            .sort((a, b) => {
+                return new Date(workflow.status.nodes[a].startedAt).getTime() - new Date(workflow.status.nodes[b].startedAt).getTime();
+            });
+
+        // Iterate over pod node ids and get their logs
+        return from(podNodeIds).pipe(
+            concatMap(nodeId => {
+                return this.getContainerLogs(workflow, container, nodeId, container, grep, archived);
```

Comment:
```suggestion
                return this.getContainerLogsFromArtifact(workflow, container, nodeId, container, grep, archived);
```

that's what I had said [above](https://github.com/argoproj/argo-workflows/pull/12591#discussion_r1517125379) and would be more accurate and efficient.
note that calling `this.getContainerLogs` actually makes these two functions mutually recursive unnecessarily.